packages/frontend/src/components/chat/ChatList.tsx

```diff
@@ -629,17 +630,21 @@ function useLogicChatPart(
     useLogicVirtualChatList(chatListIds)
 
   // effects
-  useEffect(() => {
+  const queryStrHasChanged = useHasChanged2(queryStr)
+  // TODO why don't we just call this unconditionally?????
+  // `Object.is()` will take care of infinite loops.
```

Comment:
I assume `Object.is()` is part of react's implementation of setState?

Reply:
That's right https://react.dev/reference/react/useState#setstate

> If the new value you provide is identical to the current state, as determined by an [Object.is](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) comparison, React will skip re-rendering the component and its children. This is an optimization. Although in some cases React may still need to call your component before skipping the children, it shouldn’t affect your code.

Reply:
not completely clear from the comment alone, better mention that this is done by setState

Reply:
Sure, this is just a rough draft